### DISCUSSION TITLE
Willing to pay 50 EUR if anyone can help me to fix my setup ASAP

### DISCUSSION THREAD

**Discussion by rackilucky** [teslamate-org/teslamate#2540] [Q&A]
Hello,

I need help ASAP ... it seems that my database does not talk to Teslamate although I did the setup exactly as explained and described.
My new car has 500km right now, but I want to log its whole lifetime ... so I need help ASAP !

Willing to pay 50 EUR of everything is up and running !

BR
Racky

**Comment by martinh2011** [comment]
My feeling is that a more successful approach than offering money for getting your problem solved is to provide as much information about your setup as you could, the errors you see in the logs and what you already tried to fix it yourself.

**Reply by rackilucky** [reply]
I did exactly what is described in the How-To, installed all components on my Raspberry Pi, created the YML-File with Nano, changed the passwords to something useful and started the container.
Teslamate shows a startscreen
The Grafana database seems to be not connected and does not get any data

Now I am lost ... I have no idea how to get logs or find errors as I am not an IT guy at all :(
This is why I need help and why I am willing to pay money for this help.

**Reply by martinh2011** [reply]
What OS version  is running on you pi? Try

cat /etc/os-release

What Teslamate version are you running? Click on Settings and scroll to the bottom of the page to see the version.

And by „startscreen“ you mean the homepage where the cars position and some basic data is shown like ![this one](https://github.com/adriankumpf/teslamate/blob/master/website/static/screenshots/web_interface.png)

If that is the case, the connection from teslamate to the database is probably ok and teslamate can access the Tesla API. Next step would be to look into the grafana log then via:

docker-compose logs grafana

**Reply by rackilucky** [reply]
It should be the latest raspberry and teslamate versions:

It says:

PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian

and

Version | 1.26.1
adriankumpf/teslamate

Yes, startscreen is the Teslamate-Mainscreen where I can see the car position, charging state, temperature, ...
This works fine

In the grafana logs I can see a high number of error msg:

grafana_1    | t=2022-03-12T18:52:59+0000 lvl=eror msg="Failed connecting to Postgres" logger=tsdb.postgres err="invalid option: \"harrahriikri2'\""

This is how my docker-compose looks like:

![image](https://user-images.githubusercontent.com/81877394/158134904-0b0968a9-c32d-470f-93b1-15d0b1f7b3e9.png)

**Comment by millionsofjeffries** [comment]
Remove the space between the = sign and the password on each of your password lines.

**Reply by martinh2011** [reply]
Yes, that is the problem. Postgres and Teslamate seem to be fine with a password starting with a blank, but Grafana is not. However, if you just remove the blanks you either have to reconfigure the Postgres DB credentials of the existing DB or throw away the existing db with all data via

docker-compose down -v

then change the compose file and start everything up again with

docker-compose up -d

**Reply by rackilucky** [reply]
Hehe, I think that was it - both of your comments helped me in the end to fix the issue and now its working fine !!

I would really like to pay for your efforts as offered in my original post - so if you provide your paypal accounts, that would be great

**Reply by millionsofjeffries** [reply]
Not necessary.  If anyone deserves a donation it's the developer @adriankumpf 
Glad you've got it working!

**Reply by martinh2011** [reply]
Not necessary. Either donate to the developer, @adriankumpf or donate to some organisation supporting Ukraine refugees. Or even both ;-)

**Reply by rackilucky** [reply]
Great guys, in this case I will double my bet to the benefit of Adriankumpf AND the refugees.
Thanks again for your help !

**Reply by liuguangyong93** [reply]
resolve my problems, thks
